modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php

```diff
@@ -56,65 +62,120 @@ public function handle()
         Log::info($this->description);
 
         echo "\n";
-        $this->_get_contracts();
+        $this->getContracts();
+
+        echo "\n";
+        $this->cancelEnviacontractsOnLongEndedContracts();
 
         echo "\n";
-        $this->_get_envia_contracts();
+        $this->getEnviaContracts();
+
+        echo "\n\n";
+    }
+
+    protected function cancelEnviacontractsOnLongEndedContracts()
+    {
+        foreach ($this->enviacontracts_to_be_canceled as $enviacontract) {
+            // manually and older envia contract entries
+            Log::warning(__CLASS__.": Setting state for old enviacontract $enviacontract->id to “Nicht ermittelbar”");
+            $enviacontract->state = 'Nicht ermittelbar';
+            $enviacontract->end_date = '1900-01-01';
+            $enviacontract->save();
+        }
     }
 
     /**
      * Collect all of our contracts we want to get envia TEL contracts for
      *
      * @author Patrick Reichel
      */
-    protected function _get_contracts()
+    protected function getContracts()
     {
         Log::debug(__METHOD__.' started');
 
-        $contracts = Contract::all();
+        $contracts = Contract::whereNotNull('customer_external_id')->get();
```

Comment:
This query makes it incredible faster!
```suggestion
        $contracts = Contract::whereNotNull('customer_external_id')->with('modems.mtas.phonenumbers', 'modems.mtas.phonenumbers.phonenumbermanagement')->get();
```

Reply:
fixed in e3d1968